src/crypto/sign.rs

```diff
@@ -44,33 +48,40 @@ pub enum VerifyError {
 
     #[snafu(display("invalid signature"))]
     /// The signature was invalid
-    Sodium {
+    Dalek {
         /// Error backtrace
-        backtrace: Backtrace,
+        source: ed25519_dalek::SignatureError,
     },
 }
 
 /// A public key used for verifying messages
-#[derive(
-    Copy,
-    Clone,
-    Deserialize,
-    Eq,
-    Hash,
-    PartialEq,
-    PartialOrd,
-    Ord,
-    Serialize,
-    Debug,
-)]
-pub struct PublicKey(SodiumPublicKey);
+#[derive(Copy, Clone, Eq, Debug, Serialize, Deserialize)]
+pub struct PublicKey(DalekPublicKey);
+
+impl PublicKey {
+    /// Get this `PublicKey` as a slice of bytes
+    pub fn to_bytes(self) -> [u8; PUBLICKEYBYTES] {
+        self.0.to_bytes()
+    }
```

Comment:
Why do you need that? If the user really wants that, simply use `as_bytes().clone()`.

Reply:
As opposed to `as_bytes()` this also consumes the original value as is common in Rust. 
I also find it less verbose than explicitly cloning afterwards, but that's personnal preference :)

Reply:
It's what I though but I'm not able to design a test showing consumption of the value.

This compiles and works.
```rust
    #[test]
    fn to_bytes_consumes() {
        let keypair = KeyPair::random();
        let public = keypair.public();

        let bytes = public.to_bytes();
        let bytes2 = public.to_bytes();

        assert_eq!(bytes, bytes2);
        assert_eq!(bytes, *public.as_bytes());
    }
```

Is it because an array automatically derives `Copy`?

And for patch itself, your call, you've a better rustaste than me :)

Reply:
> Is it because an array automatically derives `Copy`?

Indeed, the value is consumed by it's in fact a copy that is consumed

Reply:
Alright, thanks for the explanation :1st_place_medal: